labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +53,30 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case s:String => {
+        val len = s.length
+        s"A string with length $len"
+      }
+      case x:Int if x > 0 => {
+        "A positive integer"
+      }
+      case p:Person => {
+        val name = p.name
+        s"A person with name: $name"
+      }
+      case seq:Seq[Any] if seq.length > 10 => {
+        "Seq with more than 10 elements"
+      }
+      case seq:Seq[Any] if seq.length >= 3 => {
+        val tail = seq.drop(2)
+        val first = seq(0)
+        val second = seq(1)
+        s"first: $first, second: $second, rest: $tail"
+      }
+      case null => "A null value"
+      case _ => "Some Scala class"
+    }
```

Comment:
If you want to make this function more succinct, you don't have to declare the variables within each case. It might look better to just add them to the strings (e.g. "A string with length " + s.length). Either way, your solution definitely works!